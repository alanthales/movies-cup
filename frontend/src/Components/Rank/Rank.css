.rank-component .champions {
  background-color: #fff;
  border: 1px solid #e0e0e0;
  /* padding: 5px; */
  margin-bottom: 10px;
  display: flex;
}

.rank-component .champions .position {
  font-size: 200%;
  font-weight: bold;
  color: #fff;
  padding: 13px 17px 0px 20px;
}

.rank-component .champions .title {
  margin-left: 15px;
}

.rank-component .champions .title h4 {
  color: #333;
}
