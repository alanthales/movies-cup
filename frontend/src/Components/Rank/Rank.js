import "./Rank.css";
import React from "react";
import Box from "@material-ui/core/Box";
import Grid from "@material-ui/core/Grid";

const RankItem = ({ movie, position }) => (
  <Grid item className="champions">
    <Box bgcolor="secondary.main" className="position">{position}ยบ</Box>
    <div className="title">
      <h4>{movie.titulo}</h4>
    </div>
  </Grid>
);

const Rank = ({ champion, vice }) => (
  <Grid container direction="column" className="rank-component">
    <RankItem movie={champion} position={1} />
    <RankItem movie={vice} position={2} />
  </Grid>
);

export default Rank;