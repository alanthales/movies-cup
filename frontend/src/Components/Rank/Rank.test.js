import React from "react";
import Rank from "./Rank";
import renderer from "react-test-renderer";

test("render Rank from given props", () => {
  const champion = { titulo: "Filme vencedor" },
    vice = { titulo: "Filme vice-campe√£o" };
    
  const component = renderer.create(
    <Rank champion={champion} vice={vice} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});