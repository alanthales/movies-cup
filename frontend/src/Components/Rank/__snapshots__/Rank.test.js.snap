// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Rank from given props 1`] = `
<div
  className="MuiGrid-root rank-component MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root champions MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-105 position"
    >
      1
      º
    </div>
    <div
      className="title"
    >
      <h4>
        Filme vencedor
      </h4>
    </div>
  </div>
  <div
    className="MuiGrid-root champions MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-106 position"
    >
      2
      º
    </div>
    <div
      className="title"
    >
      <h4>
        Filme vice-campeão
      </h4>
    </div>
  </div>
</div>
`;
